### ISSUE TITLE
🐛 BUG: remove `init` from list of commands when running `npx wrangler`

### ISSUE THREAD

**Issue by rita3ko** [cloudflare/workers-sdk#5701]
### Which Cloudflare product(s) does this pertain to?

Wrangler core

### What version(s) of the tool(s) are you using?

⛅️ wrangler 3.51.2

### What operating system and version are you using?

mac

### Observed behavior
running `npx wrangler` lists `wrangler init` command, which then tells you it's deprecated

### Expected behavior
not list it

**Comment by kornelski** [comment]
`npx wrangler generate` displays the same message.

However, I without this message I would not have known that `npm create cloudflare` exists, and I would be lost not knowing how to start using Wrangler.

**Comment by irvinebroque** [comment]
Came here with same problem, but different suggestion on how we solve it.

When I run `npx wrangler@latest init` from a directory *without* a `wrangler.toml` file — `npm create cloudflare` runs — this is what should happen!

```
brendan@N64W27QG5Q test % npx wrangler@latest init

 ⛅️ wrangler 3.61.0
-------------------

Using npm as package manager.
▲ [WARNING] The `init` command is no longer supported. Please use `npm create cloudflare\@2.5.0` instead.

  The `init` command will be removed in a future version.


Running `npm create cloudflare\@2.5.0`...
```

...but when I run this from a directory that has a `wrangler.toml` already, the old `wrangler init` runs:

```
brendan@N64W27QG5Q remix-d1 % npx wrangler init
 ⛅️ wrangler 3.57.1 (update available 3.61.0)
-------------------------------------------------------
Using npm as package manager, as there is already a package-lock.json file.
▲ [WARNING] wrangler.toml already exists!


✔ Do you want to continue initializing this project? … yes
? Would you like to create a Worker at src/index.ts? › - Use arrow-keys. Return to submit.
    None
❯   Fetch handler
    Scheduled handler
```

...and if I run `wrangler generate`, in either scenario, ye ole `wrangler init` command runs:

```
brendan@N64W27QG5Q test % npx wrangler@latest generate

 ⛅️ wrangler 3.61.0
-------------------

Using npm as package manager.
▲ [WARNING] The `init` command is no longer supported. Please use `npm create cloudflare\@2.5.0` instead.

  The `init` command will be removed in a future version.


✨ Created wrangler.toml
✖ Would you like to use git to manage this Worker? … yes
```
</details>

## What should we do about it?

1. In all cases, we should run `npm create cloudflare@latest`. There should never be a case where we run the old init command runs, regardless of where you run it from.
2. We should remove the deprecation message about `wrangler init` and `wrangler generate`. It's just noise. There are [tons of projects](https://github.com/search?q=%22wrangler+generate%22&type=code) floating out there that reference the command directly. We should make things clear and easy and simple for people coming from these paths. We already hide `init` and `generate` from Wrangler's list of commands, and in the docs.
